indico/modules/events/registration/controllers/management/registration_tags.py

```diff
@@ -0,0 +1,115 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2021 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from flask import flash, redirect, request, session
+from webargs import fields
+
+from indico.core.db import db
+from indico.modules.events.registration import logger
+from indico.modules.events.registration.controllers.management import RHManageRegFormsBase
+from indico.modules.events.registration.controllers.management.reglists import RHRegistrationsActionBase
+from indico.modules.events.registration.forms import RegistrationTagForm, RegistrationTagsAssignForm
+from indico.modules.events.registration.models.tags import RegistrationTag
+from indico.modules.events.registration.views import WPManageRegistration
+from indico.util.i18n import _
+from indico.util.string import natural_sort_key
+from indico.web.args import use_kwargs
+from indico.web.flask.util import url_for
+from indico.web.util import jsonify_data, jsonify_template
+
+
+class RHManageRegistrationTags(RHManageRegFormsBase):
+    """List all registration tags for an event."""
+
+    def _process(self):
+        return WPManageRegistration.render_template('management/registration_tags.html', self.event)
+
+
+class RHRegistrationTagAdd(RHManageRegFormsBase):
+    """Add a new registration tag."""
+
+    def _process(self):
+        form = RegistrationTagForm(event=self.event)
+        if form.validate_on_submit():
+            tag = RegistrationTag(event=self.event)
+            form.populate_obj(tag, existing_only=True)
+            db.session.flush()
+            logger.info('Registration tag created by %s: %s', session.user, tag)
+            flash(_('A new tag has been created.'), 'success')
+            return jsonify_data(flash=False)
+
+        return jsonify_template('events/registration/management/registration_tag_edit.html',
+                                event=self.event, form=form, tag=None)
+
+
+class RHRegistrationTagDelete(RHManageRegFormsBase):
+    """Delete a registration tag."""
+
+    def _process(self):
+        tag = RegistrationTag.get_or_404(request.view_args['tag_id'])
```

Comment:
This allows people to delete arbitrary tags even from other events (same in the editing RH). Create an base class similar to `RHManageEventLabelBase` that queries the tag using `with_parent(self.event)` in its `_process_args` and use that as the base class for editing and deleting tags.